modules/core/src/commonMain/kotlin/fr/acinq/lightning/channel/states/Channel.kt

```diff
@@ -78,12 +78,15 @@ sealed class ChannelState {
             is Syncing -> this@ChannelState.state
             else -> this@ChannelState
         }
-        maybeSignalSensitiveTask(oldState, newState)
-        return when {
-            // we only want to fire the PaymentSent event when we transition to Closing for the first time
-            oldState is ChannelStateWithCommitments && oldState !is Closing && newState is Closing -> emitClosingEvents(oldState, newState)
-            else -> emptyList()
+        @Suppress("NAME_SHADOWING")
+        val newState = when (newState) {
+            is Offline -> newState.state
+            is Syncing -> newState.state
+            is Closed -> newState.state
+            else -> newState
         }
+        maybeSignalSensitiveTask(oldState, newState)
+        return maybeEmitClosingEvents(oldState, newState)
```

Comment:
I'm moving all the logic inside `maybeEmitClosingEvents` (it was split here and there previously).